esmvalcore/_recipe/recipe.py

```diff
@@ -631,12 +632,44 @@ def _update_extract_shape(settings, session):
     if 'extract_shape' in settings:
         shapefile = settings['extract_shape'].get('shapefile')
         if shapefile:
-            if not os.path.exists(shapefile):
-                shapefile = os.path.join(
-                    session['auxiliary_data_dir'],
-                    shapefile,
+            shapefile = Path(shapefile)
+            abs_shapefile = shapefile
+            logger.debug(
+                "extract_shape: Looking for shapefile %s", abs_shapefile
+            )
+
+            # If shapefile not found, first, try path relative to
+            # auxiliary_data_dir
+            if not abs_shapefile.exists():
+                abs_shapefile = session['auxiliary_data_dir'] / shapefile
+                logger.debug(
+                    "extract_shape: Looking for shapefile %s", abs_shapefile
+                )
+
+            # Second, try path relative to esmvalcore/preprocessor/shapefiles/
+            if not abs_shapefile.exists():
+                abs_shapefile = (
+                    Path(esmvalcore.preprocessor.__file__).parent /
+                    'shapefiles' /
+                    shapefile
+                )
+                logger.debug(
+                    "extract_shape: Looking for shapefile %s", abs_shapefile
```

Comment:
The current implementation is only convenient for those who use a recipe instead of the preprocessor function directly. From a code maintenance and as well as from a user-friendliness perspective, it would nicer if this code was part of `_area.py` and users could just call the function with the argument `shapefile='AR6'` and that would look up the correct shapefile. To add a check here (or in check.py), you could just add the word 'AR6' as a valid shapefile name.

Reply:
I agree that it would be nicer if one could use the argument `shapefile='ar6'` when calling `extract_shape` outside of running a recipe.

On the other hand, it's also nice to immediately get an error when running a recipe if the shapefile is invalid instead of getting the error after 1hr of preprocessing. I don't see a clear strategy here regarding this kind of problem when looking at other preprocessors...

Would it be an option to simply perform this check in `_recipe.py` **and** `_area.py`? I can put the code into `area.py` and then simply call it from from both locations.

Reply:
that's a very good point, Bouwe, and an even cooler point, Manu! We used to have file existence verifications in _recipe.py, back in the heady days of early 2.0 - I think there is a need for that, as well as a need for similar actions with various other files in auxialary_data_dir, so I fully support that!

Reply:
Implemented in https://github.com/ESMValGroup/ESMValCore/pull/2008/commits/cdd5bebae75d4719cc4e33ff4fbc6823c78adf71

Reply:
Thanks! There appears to be an issue with building the documentation from the docstring of `extract_shape`, could you have a look please?